Proposals/ReplaceTrustedAppsWithTags.md

```diff
@@ -0,0 +1,272 @@
+
+The user authorizes sets of tags, and the access modes to which they apply,
+for each of the apps she uses. The authorizations can be made per resource
+origin or to apply to any resource origin.
+
+The app includes the user's applicable tag authorizations when accessing the
+resource server.  The resource server verifies the authorizations and takes
+them into account when deciding whether to grant access to the resource.
+
+Terminology
+===========
+The key words "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**", "**SHALL
+NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**NOT RECOMMENDED**",
+"**MAY**", and "**OPTIONAL**" in this document are to be interpreted as
+described in BCP 14 \[[RFC2119][]\] \[[RFC8174][]\] when, and only when, they
+appear in all capitals, as shown here.
+
+Additions to WAC
+================
+This memo proposes two additions to WAC:
+
+  * `acl:app`
+  * `acl:tag`
+
+`acl:app` is similar to `acl:origin`, but specifies a prefix of an app
+identifier instead of an app origin. This allows distinguishing multiple apps
+at one origin, if desired. This memo assumes that an app identifier will be
+an OAuth2 *redirect_uri*, but this is not required. An `acl:app` matches an
+app identifier if the app identifier begins with the `acl:app`, unless the
+`acl:app` is a URI containing a fragment identifier, in which case it must
+match the app identifier exactly.
+
+The resource owner can specify one or more scope tags for an `acl:Authorization`
+instead of an `acl:origin` or `acl:app` by using `acl:tag`:
+
+	# ACL for a container of chat messages, allowing
+	# read for all authenticated users who are using
+	# an app allowing read for tag <https://vocab.example/tags#Chat>.
+	[]
+	    a acl:Authorization;
+	    acl:mode acl:Read;
+	    acl:agentClass acl:AuthenticatedAgent;
+	    acl:accessTo <./>;
+
+	    # apps the user has tagged tags:Chat or "*" for acl:Read will be allowed.
+	    acl:tag <https://vocab.example/tags#Chat>;
+
+	    acl:default <./> .
+
+If any of the tags that the user has assigned to the app that she's using for
+this access mode matches any of the tags in the `acl:Authorization`, then
+it's as if there was an `acl:app` or `acl:origin` match. Note that there is
+no universal vocabulary of tags/scopes; what tags to assign to authorizations
+is entirely at the discretion of a resource owner.
+
+App Authorization Document
+==========================
+The user associates tags for the combination of an app, resource server origin,
+and optionally the security realm (the name of the [protection space][realm];
+that is, the `realm` authentication parameter of the `WWW-Authenticate` HTTP
+response header) in an App Authorization document. Here is an example App
+Authorization document assigning tags `tags:Chat` and `tags:Pictures` to app
+`https://app.example/oauth/code` when accessing server `https://mike.example`'s
+realm `https://mike.example/auth/` for modes `acl:Read` and `acl:Write`:
+
```

Comment:
Does it mean that user have not way of granting *global* tag based access? I mean if set of tags gets used across many resource servers, does user need to create authorization for each of them?

Reply:
see at line 120 for "default app authorization", where you can specify (non-wildcard) tags for resource origin `"*"`.